File: parse.js
const fs = require('fs');
const text = fs.readFileSync('lore.txt', 'utf-8');

const events = [];
const blocks = text.split("Event ");
for (let i = 1; i < blocks.length; i++) {
    const lines = blocks[i].trim().split("\n");
    if (lines.length === 0) continue;
    
    const header = lines[0].trim();
    const match = header.match(/^(\d+):\s*(.*)/);
    if (!match) continue;
    
    const id = parseInt(match[1]);
    const title = match[2];
    
    let era = 1;
    if (id > 10) era = 2;
    if (id > 20) era = 3;
    if (id > 30) era = 4;
    
    let desc = "";
    let optA = "";
    let effA = {};
    let optB = "";
    let effB = {};
    
    let current_section = "";
    
    function parse_effects(eff_str) {
        eff_str = eff_str.replace("Efekt: [", "").replace("]", "").trim();
        const parts = eff_str.split(",");
        const parsed = {};
        for (let p of parts) {
            if (!p.includes(":")) continue;
            let [k, v] = p.split(":");
            k = k.trim();
            v = v.trim();
            let v_val = 0;
            v = v.replace(/,/g, "");
            if (v.includes("PLN")) {
                v_val = parseInt(v.replace("PLN", "").replace("+", "").replace(/ /g, ""));
            } else if (v.includes("%")) {
                v_val = parseInt(v.replace("%", "").replace("+", "").replace(/ /g, ""));
            } else {
                v_val = parseInt(v.replace("+", "").replace(/ /g, ""));
            }
            if (isNaN(v_val)) v_val = 0;
            if (k === "Street Cred") k = "StreetCred";
            parsed[k] = v_val;
        }
        return parsed;
    }

    for (let j = 1; j < lines.length; j++) {
        let line = lines[j].trim();
        if (line.startsWith("* Opis:")) {
            desc = line.replace("* Opis:", "").trim();
        } else if (line.startsWith("* Opcja A:")) {
            optA = line.replace("* Opcja A:", "").trim();
            current_section = "A";
        } else if (line.startsWith("* Opcja B:")) {
            optB = line.replace("* Opcja B:", "").trim();
            current_section = "B";
        } else if (line.startsWith("* Efekt:")) {
            const eff = parse_effects(line.replace("* Efekt:", "").trim());
            if (current_section === "A") effA = eff;
            else effB = eff;
        }
    }

    const clean = (t) => t.replace(/\d+(?=\s|$|\.)/g, '').trim();

    events.push({
        id: id,
        era: era,
        title: clean(title),
        desc: clean(desc),
        options: [
            { text: clean(optA), effects: effA },
            { text: clean(optB), effects: effB }
        ]
    });
}

const js_output = "const gameEvents = " + JSON.stringify(events, null, 4) + ";";
fs.writeFileSync("events.js", js_output, "utf-8");
console.log("Done");

File: parse.py
import re
import json

with open("lore.txt", "r", encoding="utf-8") as f:
    text = f.read()

events = []
blocks = text.split("Event ")
for block in blocks[1:]:
    lines = block.strip().split("\n")
    if not lines: continue
    header = lines[0].strip()
    match = re.match(r"(\d+):\s*(.*)", header)
    if not match: continue
    
    id = int(match.group(1))
    title = match.group(2)
    
    era = 1
    if id > 10: era = 2
    if id > 20: era = 3
    if id > 30: era = 4
    
    desc = ""
    optA = ""
    effA = {}
    optB = ""
    effB = {}
    
    current_section = ""
    
    def parse_effects(eff_str):
        eff_str = eff_str.replace("Efekt: [", "").replace("]", "").strip()
        parts = eff_str.split(", ")
        parsed = {}
        for p in parts:
            if ":" not in p: continue
            k, v = p.split(":", 1)
            k = k.strip()
            v = v.strip()
            v_val = 0
            # Some manual cleanup because of commas in numbers like 2,000
            v = v.replace(",", "")
            if "PLN" in v:
                v_val = int(v.replace("PLN", "").replace("+", "").replace(" ", ""))
            elif "%" in v:
                v_val = int(v.replace("%", "").replace("+", "").replace(" ", ""))
            else:
                try:
                    v_val = int(v.replace("+", "").replace(" ", ""))
                except:
                    v_val = 0
            if k == "Street Cred": k = "StreetCred"
            parsed[k] = v_val
        return parsed

    for line in lines[1:]:
        line = line.strip()
        if line.startswith("* Opis:"):
            desc = line.replace("* Opis:", "").strip()
        elif line.startswith("* Opcja A:"):
            optA = line.replace("* Opcja A:", "").strip()
            current_section = "A"
        elif line.startswith("* Opcja B:"):
            optB = line.replace("* Opcja B:", "").strip()
            current_section = "B"
        elif line.startswith("* Efekt:"):
            eff = parse_effects(line.replace("* Efekt:", "").strip())
            if current_section == "A":
                effA = eff
            else:
                effB = eff
                
    # Remove citation numbers from end of words
    def clean(t):
        return re.sub(r'\d+(?=\s|$|\.)', '', t).strip()

    events.append({
        "id": id,
        "era": era,
        "title": clean(title),
        "desc": clean(desc),
        "options": [
            {"text": clean(optA), "effects": effA},
            {"text": clean(optB), "effects": effB}
        ]
    })

js_output = "const gameEvents = " + json.dumps(events, indent=4, ensure_ascii=False) + ";"
with open("events.js", "w", encoding="utf-8") as f:
    f.write(js_output)
print("Done")
